diff --git a/Utils.cpp b/Utils.cpp
index 52228a65..45a15456 100644
--- a/Utils.cpp
+++ b/Utils.cpp
@@ -250,6 +250,10 @@ bool IsLegacyUserdata(const std::string& path) {
 }

 int SetQuotaProjectId(const std::string& path, long projectId) {
+    if (!IsAndroidSelinuxOperational()) {
+        return 0;
+    }
+
     struct fsxattr fsx;

     android::base::unique_fd fd(TEMP_FAILURE_RETRY(open(path.c_str(), O_RDONLY | O_CLOEXEC)));
