diff --git a/server/BandwidthController.cpp b/server/BandwidthController.cpp
index 8a967387..2581c715 100644
--- a/server/BandwidthController.cpp
+++ b/server/BandwidthController.cpp
@@ -77,6 +77,10 @@ using android::netdutils::UniqueFile;

 namespace {

+bool IsNucleusContainer() {
+    return property_get_bool("ro.nucleus.container", false);
+}
+
 const char ALERT_GLOBAL_NAME[] = "globalAlert";
 const std::string NEW_CHAIN_COMMAND = "-N ";

@@ -561,6 +565,10 @@ int BandwidthController::setGlobalAlert(int64_t bytes) {
         ALOGE("Invalid bytes value. 1..max_int64.");
         return -ERANGE;
     }
+    if (IsNucleusContainer()) {
+        mGlobalAlertBytes = bytes;
+        return 0;
+    }

     int res = 0;
     if (mGlobalAlertBytes) {
@@ -579,6 +587,10 @@ int BandwidthController::removeGlobalAlert() {

     const char *alertName = ALERT_GLOBAL_NAME;

+    if (IsNucleusContainer()) {
+        mGlobalAlertBytes = 0;
+        return 0;
+    }
     if (!mGlobalAlertBytes) {
         ALOGE("No prior alert set");
         return -1;
